src/plugins/intel_cpu/tests/unit/nodes/kernels/stack_allocator.cpp

```diff
@@ -0,0 +1,490 @@
+    EXPECT_EQ(r, 1);
+}
+
+TEST_F(StackAllocatorTest, Loop_Fail) {
+    kernel_ = [this]() {
+        Xbyak::Label l_equal;
+        Xbyak::Label l_loop;
+        Xbyak::Label l_end;
+        xor_(rcx, rcx);
+
+        StackAllocator::Transaction transaction{stack_allocator()};
+        StackAllocator::Address reg_temp0_addr{transaction, sizeof(float)};
+        StackAllocator::Address reg_temp1_addr{transaction, sizeof(int32_t)};
+        StackAllocator::Address reg_100_addr{transaction, sizeof(int32_t)};
+        StackAllocator::Address reg_200_addr{transaction, sizeof(int32_t)};
+        transaction.commit();
+        L(l_loop);
+        {
+            cmp(rcx, 10);
+            je(l_end);
+            mov(rbx.cvt32(), 100);
+            stack_mov(reg_100_addr, rbx.cvt32());
+            mov(rbx.cvt32(), 200);
+            reg_200_addr.release();
+            // NOTE: During implicit conversion to Xbyak::Address& will be thrown the exception
+            stack_mov(reg_200_addr, rbx.cvt32());
+            mov(rdx.cvt32(), reg_100_addr);
+            mov(rbx.cvt32(), 201);
+            cmp(rbx.cvt32(), rdx.cvt32());
+            mov(rax, 1);
+            add(rcx, 1);
+            jne(l_equal);
+            {
+                reg_temp1_addr.release();
+                mov(rax, 0);
+                jmp(l_loop);
+            }
+            L(l_equal);
+            {
+                jmp(l_loop);
+            }
+        }
+        L(l_end);
+    };
+    EXPECT_ANY_THROW(create_kernel<int(*) ()>());
+}
+
+TEST_F(StackAllocatorTest, Address_CheckAlignment_Fail) {
+    kernel_ = [this]() {
+        Xbyak::Label l_equal;
+        StackAllocator::Transaction transaction{stack_allocator()};
+        StackAllocator::Address byte0_addr{transaction, sizeof(int8_t)};
+        StackAllocator::Address xmm0_addr{transaction, 16};
+        transaction.commit();
+        addps(xmm0, xmm0_addr);
+    };
+    auto f = create_kernel<int(*) ()>();
+    ASSERT_DEATH({
+        f();
+    }, "");
+}
+
+template <typename T>
+class AlignedStackAllocatorTest : public StackAllocatorTest {
+public:
+    DECLARE_CPU_JIT_AUX_FUNCTIONS(AlignedStackAllocatorTest)
+
+    AlignedStackAllocatorTest()
+            : StackAllocatorTest(T::isa) {
+    }
+
+    void SetUp() override {
+    }
+
+    void TearDown() override {
+    }
+
+    void generate() override {
+        this->preamble();
+        stack_allocator_ = std::make_shared<StackAllocator>(*this, x64::cpu_isa_traits<T::isa>::vlen);
+        kernel_();
+        stack_allocator_->release();
+        stack_allocator_.reset();
+        this->postamble();
+    }
+};
+
+template<x64::cpu_isa_t Isa>
+struct IsaParam { static constexpr x64::cpu_isa_t isa = Isa; };
+
+using IsaParamTypes = ::testing::Types<IsaParam<x64::sse41>, IsaParam<x64::avx2>, IsaParam<x64::avx512_core>>;
+TYPED_TEST_SUITE(AlignedStackAllocatorTest, IsaParamTypes);
+
+TYPED_TEST(AlignedStackAllocatorTest, Address_CheckAlignment_Success) {
```

Comment:
The `TYPED_TEST` produce errors on my environment (Ubuntu 18, Debug) related to the `-Werror=suggest-override` flag. Possibly will be CI errors. You could use `TYPED_TEST_P` instead.

Reply:
I've tested using `g++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0` and in my environment everything is working properly.
Only issue I see with `REGISTER_TYPED_TEST_SUITE_P(RegPoolTest, ...`

Also looks like `intel_gpu` also uses the same approach https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/tests/module_tests/test_uqr_distribution.cpp#L58 and everything working properly.
Because `intel_cpu` and `intel_gpu` both plugins have the same host compiler, I think this issue in your environment

@OleksiiZaderykhinAltran Could you provide more detail logs ?